iemitsu.html

```diff
@@ -1,114 +1,139 @@
-document.getElementById('clock').innerHTML = '打刻';
-document.getElementById('attendance').innerHTML = '日次勤怠';
-
-// 社員情報表示
-function display(index) {
-  memberId = member[index].id;
-  memberName = member[index].name;
-};
-// 社員情報
-const member = [
-  {id:1001, name:'徳川家康'},
-  {id:1002, name:'徳川秀忠'},
-  {id:1003, name:'徳川慶喜'},
-];
-let memberId = 0;
-let memberName = ''
-display(2);//表示したい情報
-document.getElementById('timer_header').innerHTML = `${memberId} / ${memberName}`;
-
-// 現在日付表示
-const today = new Date();
-let month = today.getMonth() + 1;
-let date = today.getDate();
-let youbi = ['日', '月', '火', '水', '木', '金', '土'][today.getDay()];
-document.getElementById('today').innerHTML = (`${month}月${date}日(${youbi})`);
-
-// 現在時刻表示
-function timer(){
-  let now = new Date();
-  let hour = now.getHours();
-  let minute = now.getMinutes();
-  let second = now.getSeconds();
-  document.getElementById('time').innerHTML = `${hour}:${minute}:${second}`;
-};
-let timerID = setInterval(timer,500);
-
-// 出勤・退勤ボタン
-const startBtn = document.getElementById('start');
-const finishBtn = document.getElementById('finish');
-
-finishBtn.disabled = true;  // 初期表示
-// 出勤ボタン押下
-startBtn.addEventListener('click', function() {
-  startBtn.disabled = true;
-  finishBtn.disabled = false;
-  document.getElementById('start_record').innerHTML = '出勤：' + recordTime();
-});
-// 退勤ボタン押下
-finishBtn.addEventListener('click', function() {
-  finishBtn.disabled = true;
-  document.getElementById('finish_record').innerHTML = '退勤：' + recordTime();
-});
-// 時刻を記録
-function recordTime() {
-  let time = document.getElementById('time');
-  return time.innerHTML;
-}
-
-</script>
-
-<style>
-body { 
-  margin: 0px;
-  -webkit-text-size-adjust: 100%;
-}
-
-#iemitsu_header {
-  background-color: #e50f00;
-  height: 200px;
-}
-
-#timer {
-  text-align: center;
-}
-</style>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>IEMITSU</title>
+</head>
+<body>
+  <div>
+    <header>
+      <div id="header">
+        <div id="title"><span><h1>IEMITSU</h1>勤怠管理</span></div>
+        <div id="stamping">打刻</div>
+        <div id="attendance">日次勤怠</div>
+      </div>
+    </header>
+    <section>
+      <div id="user_info"></div>
+      <div id="timer_body">
+        <div id="today"></div>
+        <div id="current_time"></div>
+        <div id="timer_button">
+          <button id="start">出勤</button>
+          <button id="finish" disabled>退勤</button>
+        </div>
+      </div>
+    </section>
+    <div id="start_record"></div>
+    <div id="finish_record"></div>
+  </div>
+</body>
+</html>
+
+<script>
+  // 社員マスタ
+  const MST_EMPLOYEE = [
+    {id:1001, name:'徳川家康'},
+    {id:1002, name:'徳川秀忠'},
+    {id:1003, name:'徳川家光'},
+  ];
+
+  // todo:このへんの処理ってもうちょっとスマートに書けないかしら。。。

```

Comment:
わざわざ社員マスタ作る意味がないから、ユーザーとして家光だけ残していいんじゃないかな？
userIdは、`user.id`で使える。
json形式について説明したいなら別やけど...

```suggestion
  // 使用ユーザー
  const user = {id:1003, name:'徳川家光'};
```

Reply:
json形式に関しては山建のこだわりがあると思っていて、いちおう残しておいてます。
山健とも仕様検討する必要がありそう

@ymkn8crz 
このへんどうでしょうか？

今回のように社員マスタ使ってjson形式の値取得とかやるなら、下記みたいな事をしたほうが勉強になりそうだけど
いったん下記で正解つくってみる？？

・画面表示時に「id入力してね」って出てくる
・prompt関数で入力したidを取得
・マスタから名前ひっぱってくる
・名前表示する

Reply:
@HayatsuDai 
JSONは課題として実装させるorさせないにかかわらず、何かしらJSONの勉強になればいいと思っているぐらいのこだわりです。JSONはJSの基礎でもあるわけだし。

> ・画面表示時に「id入力してね」って出てくる
> ・prompt関数で入力したidを取得
> ・マスタから名前ひっぱってくる
> ・名前表示する

この提案はすごく良いと思います！！